### PR TITLE
Changes for API publish to bintray and github . Also versioning API change

### PR DESCRIPTION
**Pull request by prashanthlgowda**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -228,3 +264,64 @@ task dependencySourcesZip(type: Zip, dependsOn: 'buildSourcePathsArray' ) {
...
+    dryRun = false
+    publish = true
+    override = false
+    pkg {
+		repo = 'smi-service-dell-server-inventory'
+		name = apiVersion
+		userOrg = 'rackhd-smi'
+		desc = 'Swagger API for inventory service.'
+		licenses = ['Apache-2.0']
+		vcsUrl = 'https://rackhd-smi.bintray.com/smi-service-dell-server-inventory'
```
Comment (michaelsteven) [inline build.gradle]: Externalize this url to the gradle.properties file

File: build.gradle [hunk 2]
```diff
@@ -17,21 +19,45 @@ def _arch = 'x86_64'
...
+    def versionFile = file('src/main/resources/version.properties')
+    String versionProp = versionFile.getText('UTF-8')
+    println "updating version to '${version}' in ${versionFile}"
+    versionProp = versionProp.replaceAll(/version\.major=.*/, "version.major=${version}")
+    versionFile.write(versionProp, 'UTF-8')
+    println "updating docker tag to '${dockerTag}' in ${versionFile}"
+    versionProp = versionProp.replaceAll(/version\.tag=.*/, "version.tag=${dockerTag}")
+    versionFile.write(versionProp, 'UTF-8')
+    println "updating build number to '${build}' in ${versionFile}"
+    versionProp = versionProp.replaceAll(/version\.build=.*/, "version.build=${build}")
```
Comment (michaelsteven) [inline build.gradle]: remove the build line.  The version will include the build number

File: build.gradle [hunk 3]
```diff
@@ -228,3 +264,64 @@ task dependencySourcesZip(type: Zip, dependsOn: 'buildSourcePathsArray' ) {
...
+	key = project.hasProperty('bintrayApiKey') ? project.property('bintrayApiKey') : System.getenv('BINTRAY_API_KEY')
+    filesSpec {
+       from file(asciidoctor.outputDir.path + '/pdf/api.pdf')
+       into '.'
+    }
+    dryRun = false
+    publish = true
+    override = false
+    pkg {
+		repo = 'smi-service-dell-server-inventory'
```
Comment (michaelsteven) [inline build.gradle]: externalize this property in gradle.properties

File: build.gradle [hunk 4]
```diff
@@ -228,3 +264,64 @@ task dependencySourcesZip(type: Zip, dependsOn: 'buildSourcePathsArray' ) {
...
+		name = apiVersion
+		userOrg = 'rackhd-smi'
```
Comment (michaelsteven) [inline build.gradle]: externalize this property in gradle.properties

File: build.gradle [hunk 5]
```diff
@@ -228,3 +264,64 @@ task dependencySourcesZip(type: Zip, dependsOn: 'buildSourcePathsArray' ) {
...
+			name = apiVersion
+			desc = 'apidoc'
+			vcsTag = 'master'
+		}
+	}
+}
+
+gitPublishPush.dependsOn asciidoctor
+
+gitPublish {
```
Comment (michaelsteven) [inline build.gradle]: Don't check in this git Publish method, it has your username and we aren't using it yet

File: gradle.properties [hunk 6]
```diff
@@ -1,7 +1,11 @@
...
 dockerTag=devel
-artifactory_contextUrl=https://gtie-artifactory.us.dell.com/artifactory
+build=1
 buildInfo.licenseControl.runChecks=true
 buildInfo.licenseControl.autoDiscover=true
 buildInfo.build.name=com.dell.isg.smi:service-server-inventory
-systemProp.sonar.host.url=http://100.68.126.201:9000
\ No newline at end of file
+systemProp.sonar.host.url=http://100.68.126.201:9000
+#systemProp.org.ajoberstar.grgit.auth.username=user
```
Comment (michaelsteven) [inline gradle.properties]: remove the git lines

File: build.gradle [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 apply plugin: 'org.springframework.boot'
 apply plugin: 'java'
 apply plugin: 'eclipse'
 apply plugin: 'org.sonarqube'
 apply plugin: 'nebula.rpm'
 apply plugin: 'nebula.deb'
 apply plugin: "com.github.hierynomus.license"
+apply plugin: 'org.asciidoctor.convert'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: 'org.ajoberstar.git-publish'
```
Comment (michaelsteven) [inline build.gradle]: remove the git-publish plug-in

Comment (prashanthlgowda) [reply build.gradle]: Done

File: build.gradle (hunk 2 above)
Comment (prashanthlgowda) [reply build.gradle]: gradle.properties

Comment (prashanthlgowda) [reply build.gradle]: Done

File: build.gradle (hunk 3 above)
Comment (prashanthlgowda) [reply build.gradle]: Done

File: build.gradle (hunk 4 above)
Comment (prashanthlgowda) [reply build.gradle]: Done

File: build.gradle (hunk 5 above)
Comment (prashanthlgowda) [reply build.gradle]: Done

File: gradle.properties (hunk 6 above)
Comment (prashanthlgowda) [reply gradle.properties]: Done

**Comment by prashanthlgowda** [comment]
Done all the changes.

File: src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java [hunk 8]
```diff
@@ -56,9 +59,7 @@
 
 
     @RequestMapping(value = "/hardware", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/hardware", nickname = "hardware", notes = "This operation allow user to get complete server hardware inventory throu wsman.", response = ServerHardwareInventory.class)
-    // @ApiImplicitParams({
-    // @ApiImplicitParam(name = "credential", value = "Credential", required = true, dataType = "Credential.class", paramType = "Body", defaultValue = "no default") })
+    @ApiOperation(value = "/hardware", nickname = "hardware", notes = "This operation allow user to get complete server hardware inventory using wsman protocol.", response = ServerHardwareInventory.class)
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java]: This operation allows  @the user to retrieve a complete server software inventory via the iDRAC using the WSMan protocol.

File: src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java [hunk 9]
```diff
@@ -83,9 +84,7 @@ public ServerHardwareInventory inventory(@RequestBody Credential payload) {
 
 
     @RequestMapping(value = "/summary", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/summary", nickname = "system", notes = "This operation allow user to get partial server information throu wsman.", response = HwSystem.class)
-    // @ApiImplicitParams({
-    // @ApiImplicitParam(name = "credential", value = "Credential", required = true, dataType = "Credential.class", paramType = "Body", defaultValue = "no default") })
+    @ApiOperation(value = "/summary", nickname = "summary", notes = "This operation allow user to get server system information (System View) using wsman protocol.", response = HwSystem.class)
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java]: This operation allows the user to retrieve server system information (System View) via the iDRAC using the WSMan protocol.

File: src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java [hunk 10]
```diff
@@ -109,9 +108,7 @@ public HwSystem summary(@RequestBody Credential payload) {
 
 
     @RequestMapping(value = "/software", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/software", nickname = "software", notes = "This operation allow user to get complete server software inventory throu wsman.", response = DCIMSoftwareIdentityType.class, responseContainer = "List")
-    // @ApiImplicitParams({
-    // @ApiImplicitParam(name = "credential", value = "Credential", required = true, dataType = "Credential.class", paramType = "Body", defaultValue = "no default") })
+    @ApiOperation(value = "/software", nickname = "software", notes = "This operation allow user to get complete server software inventory using wsman protocol.", response = DCIMSoftwareIdentityType.class, responseContainer = "List")
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java]: "This operation allows the user to retrieve a complete server software inventory via the iDRAC using the WSMan protocol.

File: src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java [hunk 11]
```diff
@@ -59,7 +59,7 @@
 
 
     @RequestMapping(value = "/hardware", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/hardware", nickname = "hardware", notes = "This operation allow user to get complete server hardware inventory using WSMan protocol.", response = ServerHardwareInventory.class)
+    @ApiOperation(value = "/hardware", nickname = "hardware", notes = "This operation allow user to retrieve complete server hardware inventory via the iDRAC using WSMan protocol.", response = ServerHardwareInventory.class)
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/service/server/inventory/controller/ServerInventoryController.java]: This operation allow**s a** user to retrieve **a** complete server hardware inventory via the iDRAC using **the** WSMan protocol.

**Pull request merged**